tests/integration/product/test_product_client.py

```diff
@@ -385,3 +388,113 @@ def test__query_products_with_projection__returns_only_specified_fields(
         assert queried_product.part_number is None
         assert queried_product.keywords is None
         assert queried_product.properties is None
+
+    def test__get_products_dataframe__returns_dataframe(
+        self, client: ProductClient, create_products, unique_identifier
+    ):
+        part_number = unique_identifier
+        expected_product_dict = {
```

Comment:
```suggestion
        expected_product = {
```

nit: Meaningful names. Generally avoid using type names in variable names, if same value is represented with multiple types at a same place, then you can use the type but with prefer full names.